### PR TITLE
Fix improper response with region error

### LINKED ISSUE #921: Some requests missed to deal with unretryable RegionError
It is possible that some region error was ignored when an unretryable Error encounters in response, resulting in an incorrect empty result rather than error thrown.

Root cause for #918

### PR DESCRIPTION
**Pull request by birdstorm**
Close #921

### CODE REVIEW

**Comment by melin** [comment]
修复的代码，还是会出现下面问题：

File: tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java [hunk 1]
```diff
@@ -122,6 +123,24 @@ public ByteString get(BackOffer backOffer, ByteString key, long version)
...
   }
 
+  private boolean checkLockError(BackOffer backOffer, Kvrpcpb.KeyError error) {
+    if (error.hasLocked()) {
+      Lock lock = new Lock(error.getLocked());
+      boolean ok =
+          lockResolverClient.resolveLocks(
+              backOffer, new ArrayList<>(Collections.singletonList(lock)));
+      if (!ok) {
+        // if not resolve all locks, we wait and retry
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java]: `all locks`?  maybe there's only one lock

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java]: Maybe it is better to change to `if any lock unresolved`

File: tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java [hunk 2]
```diff
@@ -206,51 +215,57 @@ private boolean isGetSuccess(BackOffer backOffer, GetResponse resp)
...
       }
 
-      // TODO: we should retry
-      // fix me
-    }
-
-    if (resp.hasRegionError()) {
-      // TODO, we should redo the split and redo the batchGet
-      throw new RegionException(resp.getRegionError());
+      // FIXME: we should retry
```
Comment (zhexuany) [inline tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java]: Do we have a issue for this `FIXME` ?

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java]: I will create an issue for these lines.

**Comment by birdstorm** [comment]
@marsishandsome @zhexuany PTAL thx.

File: tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java (hunk 2 above)
Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java]: #927 @zhexuany

**Pull request merged**